tests/integration/test_proposals.py

```diff
@@ -0,0 +1,116 @@
+    assert response.data["submitted_at"] is not None
+    assert response.data["modified_at"] is not None
+    assert response.data["status"] == Proposal.STATUS_CHOICES.SUBMITTED
+
+    proposal_instance = Proposal.objects.get(id=response.data["id"])
+    assert str(proposal_instance) == response.data["title"]
+
+
+def test_proposal_acceptance_api(client):
+    Proposal = apps.get_model('proposals.Proposal')
+    proposal = f.create_proposal(status=Proposal.STATUS_CHOICES.SUBMITTED)
+    url = reverse('proposal-accept', kwargs={'pk': proposal.id})
+    response = client.post(url)
+    assert response.status_code == status.HTTP_401_UNAUTHORIZED
+    user = f.create_user(is_core_organizer=False)
+
+    client.login(user)
+    response = client.post(url)
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    user.is_core_organizer = True
+    user.save()
+    user.refresh_from_db()
+
+    client.login(user)
+    response = client.post(url)
+    assert response.status_code == status.HTTP_200_OK
+    proposal.refresh_from_db()
+    expected_keys = (
+        'id', 'title', 'speaker', 'status', 'kind', 'level', 'duration',
+        'abstract', 'description', 'submitted_at', 'approved_at',
+        'modified_at'
+    )
+    assert set(response.data.keys()).issubset(expected_keys)
+    assert proposal.status == Proposal.STATUS_CHOICES.ACCEPTED
+    assert proposal.approved_at is not None
+
+
+def test_proposal_retraction_api(client):
+    Proposal = apps.get_model('proposals.Proposal')
+    proposal = f.create_proposal(status=Proposal.STATUS_CHOICES.SUBMITTED)
+    url = reverse('proposal-retract', kwargs={'pk': proposal.id})
+    response = client.post(url)
+    assert response.status_code == status.HTTP_401_UNAUTHORIZED
+    user = f.create_user(is_core_organizer=False)
+
+    client.login(user)
+    response = client.post(url)
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    user.is_core_organizer = True
+    user.save()
+    user.refresh_from_db()
+
+    client.login(user)
+    response = client.post(url)
+    assert response.status_code == status.HTTP_200_OK
+    proposal.refresh_from_db()
+    expected_keys = (
+        'id', 'title', 'speaker', 'status', 'kind', 'level', 'duration',
+        'abstract', 'description', 'submitted_at', 'approved_at',
+        'modified_at'
+    )
+    assert set(response.data.keys()).issubset(expected_keys)
+    assert proposal.status == Proposal.STATUS_CHOICES.RETRACTED
+
+
+def test_proposal_listing_api(client):
+
+    no_of_test_proposals = 3
+    data = {
+        'page': 2,
+        'per_page': 1,
+    }
+    f.create_proposal(n=no_of_test_proposals)
+    url = reverse('proposal-list')
+    user = f.create_user()
+
+    client.login(user)
+    response = client.get(url, data=data)
+    assert response.status_code == status.HTTP_200_OK
+    expected_keys = (
+        'count', 'next', 'previous', 'results'
+    )
+    assert set(response.data.keys()) == set(expected_keys)
```

Comment:
You don't need to test DRF here :)

You need to test your own API; like did you get a proposal object in the listing?

Reply:
Additionally, you can check that if proposals are actually sorted by `created_at` in listing.